code/game/machinery/cryopod.dm

```diff
@@ -37,97 +37,89 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
 	GLOB.cryopod_computers -= src
 	..()
 
-/obj/machinery/computer/cryopod/attack_ai()
-	attack_hand()
+/obj/machinery/computer/cryopod/update_icon_state()
+	if(machine_stat & (NOPOWER|BROKEN))
+		icon_state = "cellconsole"
+		return ..()
+	icon_state = "cellconsole_1"
+	return ..()
 
-/obj/machinery/computer/cryopod/attack_hand(mob/user = usr)
+/obj/machinery/computer/cryopod/ui_interact(mob/user, datum/tgui/ui)
 	if(machine_stat & (NOPOWER|BROKEN))
 		return
 
-	user.set_machine(src)
 	add_fingerprint(user)
 
-	var/dat
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "CryopodConsole", name)
+		ui.open()
+
+/obj/machinery/computer/cryopod/ui_data(mob/user)
+	var/list/data = list()
+	data["allow_items"] = allow_items
+	data["frozen_crew"] = frozen_crew
+	data["frozen_items"] = list()
 
-	dat += "<hr/><br/><b>[storage_name]</b><br/>"
-	dat += "<i>Welcome, [user.real_name].</i><br/><br/><hr/>"
-	dat += "<a href='?src=[REF(src)];log=1'>View storage log</a>.<br>"
 	if(allow_items)
-		dat += "<a href='?src=[REF(src)];view=1'>View objects</a>.<br>"
-		dat += "<a href='?src=[REF(src)];item=1'>Recover object</a>.<br>"
-		dat += "<a href='?src=[REF(src)];allitems=1'>Recover all objects</a>.<br>"
+		data["frozen_items"] = frozen_items
 
-	user << browse(dat, "window=cryopod_console")
-	onclose(user, "cryopod_console")
+	var/obj/item/card/id/id_card
+	var/datum/bank_account/current_user
+	if(isliving(user))
+		var/mob/living/person = user
+		id_card = person.get_idcard()
+	if(id_card?.registered_account)
+		current_user = id_card.registered_account
+	if(current_user)
+		data["account_name"] = current_user.account_holder
```

Comment:
```suggestion
		data["card_assignment"] = "Unregistered"
		if(id_card?.assignment)
				data["card_assignment"] = id_card.assignment
```
replace 72-75 to this. Using bank as your character information should be avoided.